activesupport/lib/active_support/cache/coder.rb

```diff
@@ -0,0 +1,142 @@
+# frozen_string_literal: true
+
+require_relative "entry"
+
+module ActiveSupport
+  module Cache
+    class Coder # :nodoc:
```

Comment:
It might be worth making this public to provide a migration path from the `:coder` option to the `:serializer` option.  But that can be done in a follow-up PR.